packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -743,13 +759,32 @@ function ensureRootIsScheduled(root: FiberRoot, currentTime: number) {
         );
       }
     }
-    // The priority hasn't changed. We can reuse the existing task. Exit.
-    return;
+
+    if (
+      enableFrameEndScheduling &&
+      newCallbackPriority === DefaultLane &&
+      root.hasUnknownUpdates
+    ) {
+      // Do nothing, we need to cancel the existing default task and schedule a rAF.
```

Comment:
This seems wrong because what if a rAF were already scheduled? It would cancel it and schedule a new one, which might cause us to get shifted into a later frame